### PR TITLE
Instant feel

### LINKED ISSUE #232: Scrolling of already loaded photos is not smooth, previews get unloaded
**Describe the bug**
Scrolling up and down through the list is not smooth as previews of previously loaded pictures get unloaded. The image filetype icon is shown and the preview is loaded again, leading to flickering.

**To Reproduce**
1. Go to any view
2. Scroll down
3. Scroll back up again, and down, and up

**Expected behavior**
Scrolling is nice and smooth, you ideally never see any picture preview loading – but especially not for ones which you already scrolled past and are scrolling back up to.

**Screenshots**
Here’s a gif:
Photos scrolling jank

Using latest Photos app master.

### PR DESCRIPTION
**Pull request by Mikescops**
Implement a Service Worker to cache images and improve display of already loaded images.

Thanks a lot to @skjnldsv for the help here and there.

The gallery feels so great now!

Closes #232

### CODE REVIEW

File: lib/Controller/PageController.php [hunk 1]
```diff
@@ -91,6 +92,12 @@ public function index(): TemplateResponse {
 		Util::addStyle(Application::APP_ID, 'icons');
 
 		$response = new TemplateResponse(Application::APP_ID, 'main');
+
+		$policy = new ContentSecurityPolicy();
+		$policy->addAllowedWorkerSrcDomain("'self'");
+		$policy->addAllowedScriptDomain('blob:');
```
Comment (skjnldsv) [inline lib/Controller/PageController.php:98]: Safe enough @rullzer ?

**Comment by tacruc** [comment]
Tested and works nicely for me :+1:

File: src/components/File.vue [hunk 2]
```diff
@@ -144,8 +161,4 @@ svg {
 	width: 70%;
 	height: 70%;
 }
-
-.file--clear {
```
[the remark this answers is not in the document]
Comment (Mikescops) [reply src/components/File.vue:148]: ye

File: lib/Controller/PageController.php (hunk 1 above)
Comment (rullzer) [reply lib/Controller/PageController.php:98]: how is the script all of a sudden a blob?

Comment (Mikescops) [reply lib/Controller/PageController.php:98]: I followed this, but I may be totally wrong https://medium.com/@krishnachirumamilla/content-security-policy-worker-src-cd06ecfa2fe8

Comment (Mikescops) [reply lib/Controller/PageController.php]: @rullzer I updated it to be "self"

**Review by jancborchardt — APPROVED** [review]
Good stuff @Mikescops, no more flickering! :)

**Review by skjnldsv — APPROVED** [review]

**Comment by skjnldsv** [comment]
Please rebase and ship production files :)

File: lib/Controller/PageController.php (hunk 1 above)
Comment (rullzer) [reply lib/Controller/PageController.php]: ah ok I know why it is confusing me. Normally we only run it with nonce.
Self is a lot better yeah.
And we can always thighen it later on.

**Pull request merged**